### PR TITLE
UIQM-19 Collapse 008 field in quickMARC

### PR DESCRIPTION
**Pull request by uladislausamets**
## Purpose
https://issues.folio.org/browse/UIQM-19

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 1]
```diff
@@ -56,6 +60,32 @@ const QuickMarcEditorRows = ({
     });
   }, [moveRecord]);
 
+  const toggleFixedRow = useCallback(() => {
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:63]: it shouldn't be in QuickMarcEditorRows

**Comment by NikitaSedyx** [comment]
too many changes to collapse to add accordion to FixedField component

**Comment by uladislausamets** [comment]
> too many changes to collapse to add accordion to FixedField component
> 
> I mean it should be done in FixedField, I don't see any sense to create new Row component + have state somewhere outside and pass it through several wrappers

FixedField just component with content, wich changes on toggle, but toggle button is placed outside of this component

File: src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.js [hunk 2]
```diff
@@ -19,13 +21,13 @@ export const SUBFIELD_TYPES = {
...
 
-const renderSubField = (name, config) => {
+const renderSubField = (name, config, collapsed) => {
   const fieldName = `${name}.${config.name}`;
   const label = <FormattedMessage id={`ui-quick-marc.record.fixedField.${config.name}`} />;
 
   if (config.type === SUBFIELD_TYPES.BYTES) {
     return (
-      <FieldArray name={fieldName}>
+      <FieldArray name={fieldName} className={styles.collapsedFields}>
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.js:30]: props in column please

File: src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.css [hunk 3]
```diff
@@ -8,3 +8,16 @@
   width: 70px;
   display: inline-block;
 }
+
+.collapsedField {
+  margin-right: 10px;
+}
+
+[class*="headerWrapper"] [class*="header"] {
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.css:16]: why not to add own class?

File: src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.js [hunk 4]
```diff
@@ -80,53 +82,84 @@ const renderSubField = (name, config) => {
   );
 };
 
+const closedFields = (config, name) => {
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.js:85]: useMemo

File: src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.css (hunk 3 above)
Comment (uladislausamets) [reply src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.css:16]: This styles for accordion header, we have no way to set className directly (styles for Accordion tag isn't work)

Comment (NikitaSedyx) [reply src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.css:16]: but you defined global styles

File: src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.js [hunk 5]
```diff
@@ -19,13 +21,16 @@ export const SUBFIELD_TYPES = {
   BYTES: 'Bytes',
 };
 
-const renderSubField = (name, config) => {
+const renderSubField = useMemo(() => (name, config, collapsed) => {
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/FixedField/FixedField.js:24]: why do you changed this function when commented was for closedFIelds

**Comment by NikitaSedyx** [comment]
@uladislausamets what about tests?

File: src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js [hunk 6]
```diff
@@ -162,9 +164,12 @@ const QuickMarcEditorRows = ({
               </div>
               <div className={styles.quickMarcEditorRowContent}>
                 {
-                  recordRow.tag === '008'
-                    ? FixedFieldFactory.getFixedField(
-                      `${name}[${idx}].content`, recordRow.content.Type, recordRow.content.BLvl,
+                  withFixedField
```
Comment (NikitaSedyx) [inline src/QuickMarcEditor/QuickMarcEditorRows/QuickMarcEditorRows.js:167]: isFixedField

**Pull request merged**